plugins/plugin-docker/che-plugin-openshift-client/src/main/java/org/eclipse/che/plugin/openshift/client/OpenShiftConnector.java

```diff
@@ -606,6 +613,49 @@ public String commit(final CommitParams params) throws IOException {
     @Override
     public void getEvents(final GetEventsParams params, MessageProcessor<Event> messageProcessor) {}
 
+    @Override
+    public void getContainerLogs(final GetContainerLogsParams params, MessageProcessor<LogMessage> containerLogsProcessor)
+            throws IOException {
+        String container = params.getContainer(); // container ID
+        Pod pod = getChePodByContainerId(container);
+        if (pod != null) {
+            String podName = pod.getMetadata().getName();
+            boolean[] ret = new boolean[1];
+            ret[0] = false;
+            try (LogWatch watchLog = openShiftClient.pods().inNamespace(openShiftCheProjectName).withName(podName)
+                    .watchLog()) {
+                Watcher<Pod> watcher = new Watcher<Pod>() {
+
+                    @Override
+                    public void eventReceived(Action action, Pod resource) {
+                        if (action == Action.DELETED) {
+                            ret[0] = true;
+                        }
+                    }
+
+                    @Override
+                    public void onClose(KubernetesClientException cause) {
+                        ret[0] = true;
+                    }
+
+                };
+                openShiftClient.pods().inNamespace(openShiftCheProjectName).withName(podName).watch(watcher);
+                Thread.sleep(5000);
+                InputStream is = watchLog.getOutput();
```

Comment:
I've had some issues with output from kubernetes-client and using a reader directly. It might be worth waiting for PR #4305 and then reusing the `KubernetesOutputAdapter` class added there. You can then submit a 

```java
InputStreamPumper(watchLog.getOutput(), 
                  new KubenetesOutputAdapter(LogMessage.type.DOCKER, containerLogsProcessor));
```
and it should handle the output correctly.